arbos/programs/programs.go

```diff
@@ -333,6 +333,17 @@ func (p Programs) CodehashVersion(codeHash common.Hash) (uint16, error) {
 	return program.version, err
 }
 
+func (p Programs) CodehashSize(codeHash common.Hash) (uint32, error) {
+	program, err := p.deserializeProgram(codeHash)
+	// wasmSize represents the number of half kb units, return as bytes
+	return uint32(program.wasmSize) * 512, err
+}
+
+func (p Programs) CodehashMemoryFootprint(codeHash common.Hash) (uint16, error) {
+	program, err := p.deserializeProgram(codeHash)
+	return program.footprint, err
+}
+
```

Comment:
Let's name these `ProgramSize` and `ProgramMemoryFootprint`, but still have them take codehashes

Reply:
:+1: